File: 231_Power_Of_Two.java
class Solution {
    public boolean isPowerOfTwo(int n) {
       if(n==1)return true;
       if(n==0||n%2!=0)return false;
       if(n%2==0){
        return isPowerOfTwo(n/2);
       }
       return false;
    }
}

File: 231_Power_Of_Two.rs
impl Solution {
    pub fn is_power_of_two(n: i32) -> bool {
        if n <= 0 {
            return false;
        }
        if n == 1 {
            return true;
        }
        if n % 2 != 0 {
            return false;
        }
        Solution::is_power_of_two(n / 2)
    }
}
